File: 61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def rotateRight(self, head, k):
        """
        :type head: Optional[ListNode]
        :type k: int
        :rtype: Optional[ListNode]
        """
        if not head or not head.next or k==0:
            return head
        temp=head
        lst=[]
        while temp:
            lst.append(temp.val)
            temp=temp.next

        n = len(lst)
        k=k%n
        if k==0:
            return head
        rotate=lst[-k:]+lst[:-k]
        dummy=ListNode(0)
        t=dummy
        for x in rotate:
            t.next=ListNode(x)
            t=t.next
        return dummy.next
        

File: 61-rotate-list/rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        ListNode temp=head;
        List<Integer> list=new ArrayList<>();

        while(temp!=null){
            list.add(temp.val);
            temp=temp.next;            
        }
        // k%=list.size();
        Collections.rotate(list,k);
        ListNode dummy=new ListNode(0);
        ListNode t=dummy;
        for(int l:list){
            t.next=new ListNode(l);
            t=t.next;
        }
        return dummy.next;
    }
}